### DISCUSSION TITLE
How do I share my struggles with kons-9 without too much noise on the Discussions list?

### DISCUSSION THREAD

**Discussion by bigos** [kaveh808/kons-9#158] [Show and tell]
https://github.com/bigos/kons-9-notes/blob/main/brainstorming/merging-and-changes.org  Is this any good?

**Comment by awolven** [comment]
My advice is "patience".  The testing code is fairly new, and I am going to
be porting kons-9 to a new graphics engine soon, and the tests will surely
break then for that branch.  So there is a lot of construction going on.
The dust should start to settle in about a month or so, where we can have
better graphics and working testing all in the same branch.

**Comment by bigos** [comment]
Thank you very much for your answer. I will follow the project and will try again after a few weeks. I am patient and willing to wait to get the foundations right.

**Reply by kaveh808** [reply]
Hi Jacek,

First off, thank you for giving the system a try and testing it out.

As Andrew said, we are still making major changes in these early stages of the development, so your patience is appreciated.

I have made an attempt to keep the `main` branch working at all times. I don't understand why the demos didn't work for you.

Maybe try loading the `main` branch and working through the demos in `test/demo-kernel.lisp` by following the instructions and evaluating the expressions in each demo.

Then try the other demo files. All demos should be working.

And feel free to report your experience in this discussion. I'm happy to help out. I do acknowledge there could be more and better user documentation, and the demo files are a start.

Regards,

Kaveh

**Comment by bigos** [comment]
I had success this time. Is it possible that I have tried to run your demos before the run command?

**Reply by mikelevins** [reply]
> I had success this time. Is it possible that I have tried to run your demos before the run command?

I've done that several times myself when I come back to kons-9 after my attention has been elsewhere for a bit, so it wouldn't be surprising if that's what happened.

**Reply by kaveh808** [reply]
I will update the comments in the demo files to remind users of this.
